### ISSUE TITLE
PurchaseListener.onSuccess does not get called when purchase completes successfully with oneShotPurchaseFlow

### ISSUE THREAD

**Issue by mduong** [serso/android-checkout#43]
In successfully making a purchase using the one-shot purchase flow, `onSuccess` of the `RequestListener<Purchase>` passed to the purchase flow does not get called. However, when there's an error during the purchase process, the `onError` method does get called.

For the sake of example:
```java
ActivityCheckout mActivityCheckout;

public void purchase(final Sku sku, final RequestListener<Purchase> purchaseListener) {
    mActivityCheckout.whenReady(new Checkout.ListenerAdapter() {
        @Override
        public void onReady(BillingRequests requests) {
            requests.purchase(sku, null, mActivityCheckout.createOneShotPurchaseFlow(purchaseListener));
        }
    }
}
```

If the purchase fails in this case, the `purchaseListener`'s `onError` gets called but `onSuccess` does not get called when the purchase succeeds.

Separately creating a purchase flow that isn't a one-shot purchase flow works.

**Comment by spartako** [comment]
I can confirm this.
`PurchaseFlow.cancel()` gets called before the listener, so the call to the listener's `onSuccess` never happens.

**Comment by serso** [comment]
The fix is included in just released version 0.7.5
